### PR TITLE
helmctlr: record helm release on CR ALM-403

### PR DESCRIPTION
**Pull request by ecordell**
After a successful helm release, this writes the release back to the CR status. When the CR is loaded later, the recorded release is loaded into the in-memory release store. Tiller uses the storage to calculate differences between the previous and current release and perform actions. This allows us to wrap tiller without using the configmap storage.

One caveat is that we only store one previous release on the CR. This seems reasonable given the operator design of lostromos/SAO.

### CODE REVIEW

File: crwatcher/status.go [hunk 1]
```diff
@@ -42,48 +43,58 @@ const (
...
-		return map[string]interface{}{}
+		return nil, err
 	}
 	json.Unmarshal(jsonObj, &out)
-	return out
+	return out, nil
+}
+
+// SetPhase takes a custom resource status and returns the updated status, without updating the resource in the cluster.
+func (s *CustomResourceStatus) SetPhase(phase ResourcePhase, reason ConditionReason, message string) *CustomResourceStatus {
```
Comment (alecmerdler) [inline crwatcher/status.go:65]: I liked this better as a pure function...

**Comment by alecmerdler** [comment]
Can you push an image?

File: crwatcher/status.go (hunk 1 above)
Comment (ecordell) [reply crwatcher/status.go:65]: It's still pure, the first argument is just syntax-sugared to the left of the function name ;)

**Review by alecmerdler — APPROVED** [review]
LGTM

See it in action: https://teamui18.console.team.coreos.systems/ns/alm-403/applications/example-app-operator.v0.0.2/ExampleApp:example.com:v1alpha1/bitcoin-miner-1337/yaml

**Comment by ecordell** [comment]
@alecmerdler thanks for deploying it!

**Pull request merged**